pkg/api/types.go

```diff
@@ -172,6 +172,9 @@ type Volume struct {
 	// This is optional for now. If not specified, the Volume is implied to be an EmptyDir.
 	// This implied behavior is deprecated and will be removed in a future version.
 	VolumeSource `json:",inline,omitempty"`
+	// SecretName is the name of a Secret in the same namespace as the pod containing this volume
+	// used for volume authentication
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
Part of my concern with making secret generic is that a) you can't rely on the structure of the object to tell you whether the secret is required, b) you can't handle multiple secrets the same way (there are many possible volume plugins that may need to consume 2 or more secrets).  I'm not really in favor of this structural change.

----- Original Message -----
> > @@ -172,6 +172,9 @@ type Volume struct {
> >  	// This is optional for now. If not specified, the Volume is implied to
> >  	be an EmptyDir.
> >  	// This implied behavior is deprecated and will be removed in a future
> >  	version.
> >  	VolumeSource `json:",inline,omitempty"`
> > +	// SecretName is the name of a Secret in the same namespace as the pod
> > containing this volume
> > +	// used for volume authentication
> > +	SecretName string `json:"secretName,omitempty"`
> 
> The user is should know the secret at this time.  I don't have a use-case for
> it being late-initialized.
> 
> @markturansky There should be validation to ensure that volumes that require
> a secret have a set secretName, likewise w/ persistent volumes.
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7150/files#r29354156